src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowService.java

```diff
@@ -357,4 +361,45 @@ public void updateStage(UUID caseUUID, UUID stageUUID, Map<String, String> value
 
         camundaClient.completeTask(stageUUID, values);
     }
+
+    public ResponseEntity closeCase(UUID caseUUID) throws JsonProcessingException {
+        GetCaseworkCaseDataResponse caseDetails = caseworkClient.getCase(caseUUID);
+
+        //Get stage uuid from casework
+        String stage = caseworkClient.getActiveStage(caseDetails.getReference().replace("/", "%2F"));
+        UUID stageUUID = UUID.fromString(getStageUUID(stage));
+
+        //Mark case as complete
+        caseworkClient.completeCase(caseUUID, true);
```

Comment:
What if this fails? Would we not want this to undo?

Reply:
If this call fails then nothing will have happened so far so there shouldn't be anything to undo.

Reply:
Yeah looked at the `completeCase` method, seems relatively safe...